app/controllers/application_controller.rb

```diff
@@ -3,8 +3,153 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+    enable :sessions
+    set :session_secret, "pollywog"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect '/tweets'
+    else
+    erb :'users/login'
+    end
+  end
+
+  post '/login' do
+     user = User.find_by(username: params[:username])
+     if user && user.authenticate(params[:password])
+       session[:user_id] = user.id
+       redirect '/tweets'
+     else
+       redirect "/signup"
+     end
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+    erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == '' || params[:email] == '' || params[:password] == ''
+      redirect "/signup"
+    else
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect "/tweets"
+    end
+  end
+
+  get '/logout' do
+    if !logged_in?
+      redirect '/'
+    else
+      session.clear
+      redirect '/login'
+    end
+  end
+
+  get '/tweets' do
+    if !logged_in?
+      redirect '/login'
+    else
+    erb :'tweets/tweets'
+    end
+  end
+
+  get '/users/:slug' do
+    if !logged_in?
+      redirect '/login'
+    else
+    #  @user = User.find_by_slug(params[:slug])
+    #end
+    #@user = User.find_by_slug(params[:slug])
+    erb :'users/show'
+    end
+  end
+
+
+  get '/tweets/new' do
+    if !logged_in?
+      redirect '/login'
+    else
+      erb :'tweets/create_tweet'
+    end
+  end
+
+  post '/tweets/new' do
+    if params[:content] == ""
+      redirect '/tweets/new'
+    else
+    @tweet = Tweet.create(content: params[:content], user_id: current_user.id)
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.